bundles/CoreBundle/Migrations/Version20220120162621.php

```diff
@@ -34,64 +34,102 @@ public function up(Schema $schema): void
+            );
+        }
+
+        $this->addSql(
+            'ALTER TABLE `users_workspaces_document`
+            CHANGE `userId` `userId` int(11) unsigned NOT NULL DEFAULT \'0\';');
+
+        if (!$schema->getTable('users_workspaces_document')->hasForeignKey('fk_users_workspaces_document_users')) {
+            $this->addSql(
+                'ALTER TABLE `users_workspaces_document`
+                ADD CONSTRAINT `fk_users_workspaces_document_users`
+                FOREIGN KEY (`userId`)
+                REFERENCES `users` (`id`)
+                ON UPDATE NO ACTION
+                ON DELETE CASCADE;'
+            );
+        }
+
+
+        if (!$schema->getTable('users_workspaces_document')->hasForeignKey('fk_users_workspaces_object_objects')) {
+            $this->addSql(
+                'ALTER TABLE `users_workspaces_object`
+                ADD CONSTRAINT `fk_users_workspaces_object_objects`
+                FOREIGN KEY (`cid`)
+                REFERENCES `objects` (`o_id`)
+                ON UPDATE NO ACTION
+                ON DELETE CASCADE;'
+            );
+        }
+
+        $this->addSql(
+            'ALTER TABLE `users_workspaces_object`
+            CHANGE `userId` `userId` int(11) unsigned NOT NULL DEFAULT \'0\';');
+
+        if (!$schema->getTable('users_workspaces_document')->hasForeignKey('fk_users_workspaces_object_users')) {
+            $this->addSql(
+                'ALTER TABLE `users_workspaces_object`
+                ADD CONSTRAINT `fk_users_workspaces_object_users`
+                FOREIGN KEY (`userId`)
+                REFERENCES `users` (`id`)
+                ON UPDATE NO ACTION
+                ON DELETE CASCADE;'
+            );
+        }
 
         //enable foreign key checks
         $this->addSql('SET foreign_key_checks = 1');
     }
 
     public function down(Schema $schema): void
     {
-        $this->addSql('ALTER TABLE `users_workspaces_asset`
-                CHANGE `userId` `userId` int(11) NOT NULL DEFAULT \'0\',
-                DROP FOREIGN KEY IF EXISTS `fk_users_workspaces_asset_assets`,
-                DROP FOREIGN KEY IF EXISTS `fk_users_workspaces_asset_users`');
-
-        $this->addSql('ALTER TABLE `users_workspaces_document`
-                CHANGE `userId` `userId` int(11) NOT NULL DEFAULT \'0\',
-                DROP FOREIGN KEY IF EXISTS `fk_users_workspaces_document_documents`,
-                DROP FOREIGN KEY IF EXISTS `fk_users_workspaces_document_users`');
-
-        $this->addSql('ALTER TABLE `users_workspaces_object`
-                CHANGE `userId` `userId` int(11) NOT NULL DEFAULT \'0\',
-                DROP FOREIGN KEY IF EXISTS `fk_users_workspaces_object_objects`,
-                DROP FOREIGN KEY IF EXISTS `fk_users_workspaces_object_users`');
+        foreach(['asset', 'document', 'object'] as $elementType) {
+            $this->addSql('ALTER TABLE `users_workspaces_'.$elementType.'` CHANGE `userId` `userId` int(11) NOT NULL DEFAULT \'0\'');
```

Comment:
this line needs to be moved after the foreign key drop

Reply:
Fixed. I actually kept the order of the original statement but obviously MySQL is clever enough to reorder this composite statement (or if not, this never worked 😄  ).

Reply:
Apparently doesn't work like that, at least it seem it behaves differently when calling in separately queries, it has to follow the order they are added/executed.

![image](https://user-images.githubusercontent.com/6014195/220841696-b6ec2efc-cd3c-42ef-bf21-5d62cfece6f7.png)
Alternatively, it could be bypassed by using the same toggle of `SET foreign_key_checks` in the `migration up()`